5.0 Draft/IDD/IDDs Authorization System/eu.arrowhead.authorization-management-http-json.yml

```diff
@@ -0,0 +1,518 @@
+openapi: 3.0.3
+info:
+  title: Authorization Policy Management API
+  description: |
+    This specification outlines how to realize the _Arrowhead Authorization Management_ service on top of either of the HTTP or HTTPS
+    protocols with payload encoded in JSON. The service allows for resources, data owners, service providers, or system administrators
+    to manage and express authorization rules to resources. 
+
+    To express authorization policies over resources needing protection, a _resource_ object must be created in the policy engine. This 
+    newly created resource can be _associated_ to existing attributes using the _/management/associations_ endpoint. The same applies
+    whenever a new _subject_ wanting to consume a _resource_ enters to the environment. A _subject_ object must be created in the policy
+    engine, and it needs to be connected to an attribute using the _/management/associations_ endpoint. This association will connect
+    the newly created _resource_ or _subject_ to existing authorization policies. 
+
+
+  version: 1.0.0
+  # Additional information present in the IDD template
+  x-transfer-protocol: [HTTP, HTTPS]
+  x-encryption-method: to-add
+  x-encoding: JSON
+  x-compression: to-add
+  x-semantics: to-add
+  x-Ontology: to-add
+
+paths:
+  /management/associations:
+    post:
+      summary: Create a New association
+      description: Create a new association relating resources to resource-attributes, subjects (consumers) to 
+        subject attributes, or attribute-attribute associations. 
+        The resource or subject must be an existing object. The specified attribute can be one existing from the 
+        attributes collection, or if a non-existant attribute is specified, the new attribute will be created in 
+        the collection. 
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Association' 
+            examples: 
+              example1:
+                summary: A call to associate a Temperature sensor service to the existing "OutsideTemp" resource attribute. 
+                value: 
+                  id: "123123"
+                  association-type: "resource"
+                  target:
+                    id: "existing-resource123"
+                    name: "eu.arrowhead.tempreadings-http-json"
+                  attribute:
+                    id: "existing-attribute1233"
+                    name: "OutsideTemp"
+                    attribute-type: "resource"
+      responses:
+        '201':
+          description: Association created successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Association'
+
+    get:
+      summary: List Policies
+      description: Retrieve a list of all policies.
```

Comment:
This is kind of dangerous, as this endpoint could be used to mount a Denial-of-Service attack. An attacker with access to this endpoint can simply send a request to it over and over again and ignore the response to bring down the system. The way you mitigate this kind of attack is by setting a limit to how many items can be requested, and allowing to consumer to request only some items via some kind of paging mechanism. In the Service Discovery IDD, the offset/limit paging mechanism is used. Note, also, how a `Limit` header is included in the responses to relevant requests to help the consumer know the maximum limit is allowed to include in its requests.